android/src/main/java/com/ioreactnativeiso18013/IoReactNativeIso18013Module.kt

```diff
@@ -398,44 +370,24 @@ class IoReactNativeIso18013Module(reactContext: ReactApplicationContext) :
     */
   }
 
-  private enum class ModuleException(
-    val ex: Exception
-  ) {
-    /** ISO18013-5 related errors **/
-    DRH_NOT_DEFINED(Exception("DRH_NOT_DEFINED")),
-    QR_ENGAGEMENT_NOT_DEFINED_ERROR(Exception("QR_ENGAGEMENT_NOT_DEFINED_ERROR")),
-    START_ERROR(Exception("START_ERROR")),
-    GET_QR_CODE_ERROR(Exception("GET_QR_CODE_ERROR")),
-    GENERATE_RESPONSE_ERROR(Exception("GENERATE_RESPONSE_ERROR")),
-    SEND_RESPONSE_ERROR(Exception("SEND_RESPONSE_ERROR")),
-    SEND_ERROR_RESPONSE_ERROR(Exception("SEND_ERROR_RESPONSE_ERROR")),
-    CLOSE_ERROR(Exception("CLOSE_ERROR")),
-
-    /** ISO18013-7 related errors **/
-    UNABLE_TO_GENERATE_RESPONSE(Exception("UNABLE_TO_GENERATE_RESPONSE")),
-    UNABLE_TO_GENERATE_TRANSCRIPT(Exception("UNABLE_TO_GENERATE_TRANSCRIPT")),
-    INVALID_DOC_REQUESTED(Exception("INVALID_DOC_REQUESTED")),
-    GENERATE_OID4VP_DEVICE_RESPONSE_FAILED(Exception("GENERATE_OID4VP_DEVICE_RESPONSE_FAILED"));
-
-    fun reject(
-      promise: Promise, vararg args: Pair<String, String>
-    ) {
-      exMap(*args).let {
-        promise.reject(it.first, ex.message, it.second)
-      }
-    }
-
-    private fun exMap(vararg args: Pair<String, String>): Pair<String, WritableMap> {
-      val writableMap = WritableNativeMap()
-      args.forEach { writableMap.putString(it.first, it.second) }
-      return Pair(this.ex.message ?: "UNKNOWN", writableMap)
-    }
-  }
-
-  private fun getExceptionMessageOrEmpty(e: Exception): String = e.message ?: ""
-
   companion object {
     const val NAME = "IoReactNativeIso18013"
-    const val ERROR_KEY = "error"
+    const val NOT_INITIALIZED_ERROR_MESSAGE = "Resources not initialized properly, call the start method before this one."
+
+    // Errors which this module uses to reject a promise
+    private object ModuleErrorCodes {
+      // ISO18013-5 related errors
+      const val DRH_NOT_DEFINED = "DRH_NOT_DEFINED"
+      const val QR_ENGAGEMENT_NOT_DEFINED = "QR_ENGAGEMENT_NOT_DEFINED"
+      const val START_ERROR = "START_ERROR"
+      const val GET_QR_CODE_ERROR = "GET_QR_CODE_ERROR"
+      const val GENERATE_RESPONSE_ERROR = "GENERATE_RESPONSE_ERROR"
+      const val SEND_RESPONSE_ERROR = "SEND_RESPONSE_ERROR"
+      const val SEND_ERROR_RESPONSE_ERROR = "SEND_ERROR_RESPONSE_ERROR"
+      const val CLOSE_ERROR = "CLOSE_ERROR"
+
+      // ISO18013-7 related errors
+      const val GENERATE_OID4VP_RESPONSE_ERROR = "GENERATE_RESPONSE_ERROR"
```

Comment:
```suggestion
      const val GENERATE_OID4VP_RESPONSE_ERROR = "GENERATE_OID4VP_RESPONSE_ERROR"
```

By looking at the typescript code, I think this const value should be renamed as suggested, correct me if I'm wrong.